engine/message/db.go

```diff
@@ -763,65 +779,39 @@ func (db *DB) updateStuckMessages(ctx context.Context, statusFn StatusFunc) erro
 	return nil
 }
 
-const workersPerType = 5
-
 func (db *DB) sendMessagesByType(ctx context.Context, cLock *processinglock.Conn, send SendFunc, q *queue, typ notification.DestType) error {
-	limit := db.c.RateLimit[typ]
-
-	toSend := int(limit.Batch.Seconds() * float64(limit.PerSecond))
-	var failures, processing int
-	sentCount := func() int {
-		return q.SentByType(typ, limit.Batch) - failures
-	}
-
-	sendCh := make(chan *Message, workersPerType)
-	errCh := make(chan error, workersPerType)
-	resCh := make(chan bool, workersPerType)
-	for i := 0; i < workersPerType; i++ {
+	ch := make(chan error)
+	var count int
+	for {
+		msg := q.NextByType(typ)
+		if msg == nil {
+			break
+		}
+		count++
 		go func() {
-			for msg := range sendCh {
-				sent, err := db.sendMessage(ctx, cLock, send, msg)
-				if err != nil {
-					errCh <- err
-				} else {
-					resCh <- sent
-				}
-			}
+			_, err := db.sendMessage(ctx, cLock, send, msg)
+			ch <- err
 		}()
 	}
-	defer close(sendCh)
 
-	var done bool
-	for {
-		if !done && sentCount() < toSend {
-			msg := q.NextByType(typ)
-			if msg == nil {
-				done = true
+	var failed bool
```

Comment:
It's preferred to use `var` with the type instead of `:=` when using the zero value (e.g., `false`, `0`).

```go
var failed bool
var count int
var buf bytes.Buffer

// vs.
failed := false
count := 0
buf := bytes.Buffer{}
// etc...
```